pkg/core/types.go

```diff
@@ -42,3 +42,17 @@ type Object interface {
 	runtime.Object
 	metav1.Object
 }
+
+// ConfigField is used to contain information about a specific field within
+// a ConfigMap/Secret to control how Wave uses that field.
+type ConfigField struct {
```

Comment:
If this only has private fields, does it need to be public? Where is it being used outside of this package?

Reply:
Good point - will change.